### PR TITLE
Closes #8 | Create dataset loader for MLQA

### LINKED ISSUE #8: Create dataset loader for MLQA
Dataloader name: `mlqa/mlqa.py`
Data catalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?mlqa

|     Dataset        | mlqa  |
|-------------|---|
| Description | MLQA (MultiLingual Question Answering) is a benchmark dataset for evaluating cross-lingual question answering performance. MLQA consists of over 5K extractive QA instances (12K in English) in SQuAD format in seven languages - English, Arabic, German, Spanish, Hindi, Vietnamese and Simplified Chinese. MLQA is highly parallel, with QA instances parallel between 4 different languages on average. |
| Subsets | mlqa-translate-test.vi, mlqa-translate-train.vi, mlqa.vi.ar, mlqa.vi.de, mlqa.vi.vi, mlqa.vi.zh, mlqa.vi.en, mlqa.vi.es, mlqa.vi.hi, mlqa.ar.vi, mlqa.de.vi, mlqa.zh.vi, mlqa.en.vi, mlqa.es.vi, mlqa.hi.vi |
| Languages | vie |
| License | Creative Commons Attribution Non Commercial 4.0 (cc-by-nc-4.0) |
| Homepage | https://github.com/facebookresearch/MLQA |
| HF URL | https://huggingface.co/datasets/mlqa |
| Paper URL | https://aclanthology.org/2020.acl-main.653 |

### PR DESCRIPTION
**Pull request by IvanHalimP**
Closes #8 
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        
+        if (name_split[0].startswith("mlqa-translate-train")):
+            url = _URL+_TRANSLATE_TRAIN_URL
+            data_path = dl_manager.download(url)
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # Whatever you put in gen_kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": f"{name_split[0]}/vi_squad-translate-train-train-v1.1.json",
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mlqa/mlqa.py:128]: `name_split[0]` here contains `mlqa-translate-train.vi`
but the expected filepath should be `mlqa-translate-train` only, and the `.vi` part need to be stripped

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+               "mlqa.vi.es", 
+               "mlqa.vi.hi", 
+               "mlqa.vi.vi",
+               "mlqa.ar.vi", 
+               "mlqa.de.vi", 
+               "mlqa.zh.vi", 
+               "mlqa.en.vi", 
+               "mlqa.es.vi", 
+               "mlqa.hi.vi",
+               "mlqa"]
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mlqa/mlqa.py:65]: this `mlqa` subset is not handled in `_split_generators`,
maybe we should remove this?

**Review by ryanignatius — CHANGES_REQUESTED** [review]

**Comment by ryanignatius** [comment]
please add `__init.py__` file too

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+                                "question": question,
+                                "answer": answers_text,
+                                "id": id_,
+                                "choices": [],
+                                "type": "extractive",
+                                "document_id": count
+                            }
+                            count+=1
+                        else:
+                            raise ValueError(f"Invalid config: {self.config.name}")
```
[the remark this answers is not in the document]
Comment (IvanHalimP) [reply seacrowd/sea_datasets/mlqa/mlqa.py:223]: okay, you mean that we need 2 if(s) one for source and another for seacrowd_qa, then both cases can be ended with else: raise ValueError...(etc) right?

**Comment by IvanHalimP** [comment]
Hi, I've modified the mlqa.py file and added __init__.py file.

**Review by SamuelCahyawijaya — APPROVED** [review]
LGTM!

File: seacrowd/sea_datasets/mlqa/mlqa.py (hunk 2 above)
Comment (IvanHalimP) [reply seacrowd/sea_datasets/mlqa/mlqa.py]: done

File: seacrowd/sea_datasets/mlqa/mlqa.py (hunk 1 above)
Comment (IvanHalimP) [reply seacrowd/sea_datasets/mlqa/mlqa.py]: done

File: seacrowd/sea_datasets/mlqa/mlqa.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (IvanHalimP) [reply seacrowd/sea_datasets/mlqa/mlqa.py]: done

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+        if name_split[0].startswith("mlqa-translate-train"):
+            config_name, lang = name_split[0].split(".")
+            url = _URL + _TRANSLATE_TRAIN_URL
+            data_path = dl_manager.download(url)
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    # Whatever you put in gen_kwargs will be passed to _generate_examples
+                    gen_kwargs={
+                        "filepath": f"{config_name}.{lang}/{lang}_squad-translate-train-train-v1.1.json",
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mlqa/mlqa.py:123]: the expected folder is `mlqa-translate-train` only so the lang is not needed in the folder part
```suggestion
                        "filepath": f"{config_name}/{lang}_squad-translate-train-train-v1.1.json",
```

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                        "files": dl_manager.iter_archive(data_path),
+                        "split": "train",
+                    },
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={
+                        "filepath": f"{config_name}.{lang}/{lang}_squad-translate-train-dev-v1.1.json",
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mlqa/mlqa.py:131]: the expected folder is `mlqa-translate-train` only so the lang is not needed in the folder part
```suggestion
                        "filepath": f"{config_name}/{lang}_squad-translate-train-dev-v1.1.json",
```

File: seacrowd/sea_datasets/mlqa/mlqa.py [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+        elif name_split[0].startswith("mlqa-translate-test"):
+            config_name, lang = name_split[0].split(".")
+            url = _URL + _TRANSLATE_TEST_URL
+            data_path = dl_manager.download(url)
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={
+                        "filepath": f"{config_name}.{lang}/translate-test-context-{lang}-question-{lang}.json",
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/mlqa/mlqa.py:146]: the expected folder is `mlqa-translate-test` only so the lang is not needed in the folder part
```suggestion
                        "filepath": f"{config_name}/translate-test-context-{lang}-question-{lang}.json",
```

File: seacrowd/sea_datasets/mlqa/mlqa.py (hunk 5 above)
Comment (ryanignatius) [reply seacrowd/sea_datasets/mlqa/mlqa.py:131]: just missing one more 👀

**Review by ryanignatius — APPROVED** [review]

**Pull request merged**